lib/cocoapods/installer/analyzer.rb

```diff
@@ -429,59 +433,33 @@ def generate_target(target_definition, pod_targets)
           end
         end
 
-        target.pod_targets = filter_pod_targets_for_target_definition(pod_targets, target_definition)
+        target.pod_targets = filter_pod_targets_for_target_definition(target_definition, pod_targets)
 
         target
       end
 
       # Returns a filtered list of pod targets that should or should not be part of the target definition. Pod targets
       # used by tests only are filtered.
       #
-      # @param [Array<PodTarget>] pod_targets
-      #        the array of pod targets to check against
-      #
       # @param [TargetDefinition] target_definition
       #        the target definition to use as the base for filtering
       #
-      # @return [Array<PodTarget>] the filtered list of pod targets.
-      #
-      def filter_pod_targets_for_target_definition(pod_targets, target_definition)
-        pod_targets.select do |pod_target|
-          pod_target.target_definitions.include?(target_definition) && !pod_target_test_only?(pod_target, pod_targets)
-        end
-      end
-
-      # Returns true if a pod target is only used by other pod targets as a test dependency and therefore should
-      # not be included as part of the aggregate target.
-      #
-      # @param [PodTarget] pod_target
-      #        the pod target being queried
-      #
       # @param [Array<PodTarget>] pod_targets
       #        the array of pod targets to check against
       #
-      # @return [Boolean] if the pod target is only referenced from test dependencies.
+      # @return [Array<PodTarget>] the filtered list of pod targets.
       #
-      def pod_target_test_only?(pod_target, pod_targets)
```

Comment:
Method was removed and so did the caching. The resolver now takes care of this much better.